CMakeLists.txt

```diff
@@ -1,11 +1,36 @@
 # CMake4GDAL project is distributed under MIT license. See accompanying file LICENSE.txt.
+# Modified by Mi Zhang, LuoJiaNET Development and Research Group, Wuhan University, 2022
+
 cmake_minimum_required(VERSION 3.9...3.23)
 
 project(gdal LANGUAGES C CXX)
 include(CTest)
 
 set(GDAL_LIB_TARGET_NAME GDAL)
 
+###############################################################################
+# Generate setup.cfg
+###############################################################################
+if (WIN32)
+    set(GDAL_COMPILER mingw32)
+elseif(UNIX OR APPLE)
+    set(GDAL_COMPILER unix)
+else ()
+    message(FATAL_ERROR
```

Comment:
instead of erroring out, I would just not define the "compiler = ..." line in setup.cfg
You could use a @PYTHON_COMPILER@ placeholder in the template and only set it to non empty when you want to override, which is probably only for mingw32